modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java

```diff
@@ -1536,6 +1574,41 @@ public void setBinaryMetadataFileStoreDir(@Nullable File binaryMetadataFileStore
         this.binaryMetadataFileStoreDir = binaryMetadataFileStoreDir;
     }
 
+    /** {@inheritDoc} */
+    @Override public void removeType(int typeId) {
+        if (metadataLocCache.computeIfAbsent(typeId,
+                (id) -> {
+                    throw new IgniteException("Failed to remove metadata, type not found: " + id);
+                })
+            .removing())
+            throw new IgniteException("Failed to remove metadata, type is being removed: " + typeId);
```

Comment:
This looks hard to understand.

Reply:
Let's add a comment like this:
Fail if "typeId" wasn't found otherwise check "removing" flag.

Reply:
Simplified